core/src/test/java/ai/timefold/solver/core/api/solver/SolverManagerTest.java

```diff
@@ -263,6 +264,173 @@ void solveGenerics() throws ExecutionException, InterruptedException {
         solverJob.getFinalBestSolution();
     }
 
+    @Test
+    @Timeout(60)
+    void solveWithFirstInitializedSolutionConsumer() throws ExecutionException, InterruptedException {
```

Comment:
This test is actually several independent tests in one; only CH, CH-LS, etc. Let's split them.